SU-SWS/stanford_components
# EARTH-92 EARTH-243 deep banner

# READY FOR REVIEW

# Summary
- Moved fields to variants
- Changed markup to use attributes
- created unique id's to each pattern
- Added `get_param` twig function to get a `$_GET` parameter. useful for no javascript work
- used twig functions we merged in
- Removed text regions that were just link titles since we will use link fields.
- Made deep linking banner working without javascript
- Compatabile rendering with fields
- improved expandable banner javascript to use aria-expanded & workable with fields.
- Created md5 filter/function. not using now, but maybe ¯\\_(ツ)_/¯
- changed field `cards` to `items` to be more consistent.

# Note
- with the attributes work, some css styles are clashing from the theme.
- The unique id's doesn't work on /patterns page because of the `include '@stanford_components/....html.twig' with item_vars` This doesn't call the preprocess hook and so they don't get the unique id. But this shouldn't be an issue on entity pages since this `include` part isn't called.

# Needed By (Date)
- End of spring

# Urgency
- Somewhat

# Steps to Test

1. Checkout branch
2. `drush cr`
3. checkout `EARTH-92-deep-banner` branch of [stanford_paragraph_types](https://github.com/SU-SWS/stanford_paragraph_types/pull/43)
4. `drush en stanford_deep_linking_banner -y`
5. create a new complex page using the deep linking banner
6. Add 2 expandable cards with links
7. save the node.
8. inspect the patterns to ensure each has a unique id on the outer div. 
9. turn off javascript, refresh page
10. click on the expandable banner open button
11. page should reload with the card expanded and the focus on the card.
12. also test focus doesnt get lost.

# Associated Issues and/or People
- https://github.com/SU-SWS/stanford_paragraph_types/pull/43
- https://github.com/SU-SWS/stanford_components/pull/44

# See Also
- [deep linking banner](https://github.com/SU-SWS/stanford_paragraph_types/pull/43)
